src/prefect/agent/kubernetes/agent.py

```diff
@@ -103,9 +105,177 @@ def __init__(
+        )
+
+        more = True
+        _continue = ""
+        while more:
+            try:
+                jobs = self.batch_client.list_namespaced_job(
+                    namespace=self.namespace,
+                    label_selector="prefect.io/identifier",
+                    limit=20,
+                    _continue=_continue,
+                )
+                _continue = jobs.metadata._continue
+                more = bool(_continue)
+
+                for job in jobs.items:
+                    pods = self.core_client.list_namespaced_pod(
+                        namespace=self.namespace,
+                        label_selector="prefect.io/identifier={}".format(
+                            job.metadata.labels.get("prefect.io/identifier")
+                        ),
+                    )
+
+                    pod_names = []
+                    for pod in pods.items:
+                        pod_names.append(pod.metadata.name)
+
+                    jobs_list.append(
+                        {
+                            "job_name": job.metadata.name,
+                            "pod_names": pod_names,
+                            "flow_run_id": job.metadata.labels.get(
+                                "prefect.io/flow_run_id"
+                            ),
+                        }
+                    )
+            except self.k8s_client.rest.ApiException as exc:
+                if exc.status == 410:
+                    self.logger.debug("Refreshing job listing token...")
+                    _continue = ""
+                    continue
+
+        self.logger.debug(
+            f"Found {len(jobs_list)} existing prefect jobs in the cluster"
+        )
+        return jobs_list
+
+    def _read_prefect_jobs(self, jobs: list) -> None:
+        """
+        This function checks if jobs are `Failed` or `Succeeded` and if they are then the jobs are
+        deleted from the namespace. If one of the job's pods happen to run into image pulling errors
+        then the flow run is failed and the job is still deleted.
+
+        Args:
+            jobs: A list of job identifiers from Kubernetes
+        """
+        self.logger.debug(f"Reading statuses for {len(jobs)} jobs...")
+        for job in jobs:
+            delete_job = False
+
+            try:
+                job_status = self.batch_client.read_namespaced_job_status(
+                    namespace=self.namespace,
+                    name=job["job_name"],
+                ).status
+            except self.k8s_client.rest.ApiException as exc:
+                self.logger.error(
+                    f"{exc.status} error attempting to read status of job {job['job_name']}"
+                )
+                continue
+
+            if not delete_job and not job_status.failed and not job_status.succeeded:
+                for pod_name in job["pod_names"]:
+                    try:
+                        pod_status = self.core_client.read_namespaced_pod_status(
```

Comment:
Same here, we should already have the pod status as part of `list_namespaced_pod` above.